src/oc/electron/main.cljs

```diff
@@ -158,4 +160,7 @@
   (.on ipc-main "set-badge-count" (fn [event arg] (.setBadgeCount app arg)))
   (.on ipc-main "show-desktop-window" (fn [event arg] (when @main-window
                                                         (.show @main-window))))
+  (.on ipc-main "window-has-focus?" (fn [event]
+                                      (let [ret-value (boolean (.getFocusedWindow BrowserWindow))]
+                                        (set! (.-returnValue event) ret-value))))
```

Comment:
Let me see if i get this.
When you call `ipcRenderer.sendSync('window-has-focus?')` it generates an event.
Then here you capture that event and add a boolean value to it that says if the window is focused, that correct?

Reply:
@bago2k4 Yeah, the renderer process generates a synchronous event (blocks the UI), and the main process catches that event above, and sets the `returnValue` field on the event to the focus state of the window. The `returnValue` field is how Electron responds to synchronous events.